end-to-end-tests/src/helpers/ctx.rs

```diff
@@ -179,15 +216,11 @@ pub async fn build_client() -> Result<oxide_client::Client> {
             // codes.  See progenitor#451.
             eprintln!("{}: attempting to log into API", Utc::now());
 
-            let mut builder = reqwest::ClientBuilder::new()
+            let builder = client_params
+                .reqwest_builder()
```

Comment:
btw, I got rid of the 303 redirect on local login in #3217, so you might not need the raw reqwest builder anymore. The SAML endpoint still 303s on success, but that one is more plausibly necessary and it's not being hit by the API client.

Reply:
That's cool.  Unwinding this will be a bit more work (there are two callers, each setting up reqwest clients for this) so I'd like to defer it from this PR.